packages/expo-asset/src/Asset.js

```diff
@@ -193,6 +232,34 @@ export default class Asset {
     return asset;
   }
 
+  static fromURI(uri) {
+    const metaHash = uri;
+    if (Asset.byHash[metaHash]) {
```

Comment:
I think it'd be preferable to use the system's caching policy (which should honor HTTP cache headers) and leave this hash-based caching only for Expo-hosted assets where we can make assumptions about the caching policies.